plugin/rewrite/edns0.go

```diff
@@ -75,24 +72,19 @@ Option:
 	return result
 }
 
-// Mode returns the processing mode
-func (rule *edns0NsidRule) Mode() string {
-	return rule.mode
-}
+// Mode returns the processing mode.
+func (rule *edns0NsidRule) Mode() string { return rule.mode }
 
 // GetResponseRule return a rule to rewrite the response with. Currently not implemented.
-func (rule *edns0NsidRule) GetResponseRule() ResponseRule {
-	return ResponseRule{}
-}
+func (rule *edns0NsidRule) GetResponseRule() ResponseRule { return ResponseRule{} }
 
-// Rewrite will alter the request EDNS0 local options
-func (rule *edns0LocalRule) Rewrite(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) Result {
+// Rewrite will alter the request EDNS0 local options.
+func (rule *edns0LocalRule) Rewrite(state request.Request) Result {
 	result := RewriteIgnored
-	o := setupEdns0Opt(r)
+	o := setupEdns0Opt(state.Req)
 	found := false
 	for _, s := range o.Option {
-		switch e := s.(type) {
-		case *dns.EDNS0_LOCAL:
+		if e, ok := s.(*dns.EDNS0_LOCAL); ok {
```

Comment:
here same refactor with using break and returns could happen.

Reply:
done